machiavelli-cards/src/main/java/pl/jellytech/machiavelli/cards/DataSourceConfig.java

```diff
@@ -0,0 +1,43 @@
+package pl.jellytech.machiavelli.cards;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import javax.sql.DataSource;
+import javax.xml.crypto.Data;
+import java.sql.SQLException;
+
+@Configuration
+public class DataSourceConfig extends com.zaxxer.hikari.HikariConfig {
+
+    @Value("${machiavelli.card.datasource.url}")
+    private String url;
+    @Value("${machiavelli.card.datasource.driverClassName}")
+    private String driverClassName;
+    @Value("{machiavelli.card.datasource.username}")
+    private String username;
+    @Value("{machiavelli.card.datasource.password}")
+    private String password;
+
+
+    @Bean
+    @ConfigurationProperties(prefix ="machiavelli.card.datasource")
+    public HikariConfig hikariConfig() {
+        HikariConfig config = new HikariConfig();
+        config.setJdbcUrl(this.url);
+        config.setDriverClassName(this.driverClassName);
+        config.setUsername(this.username);
+        config.setPassword(this.password);
```

Comment:
good, first time I see a class extending HikariConfig - isn't really necessary but still nice to know we can do it this way
Just add additional properties here - set the particular timeouts that can be invoked. That's the main beauty of using HikariConfig - it allows for additional configuration

Reply:
unresolving - connection timeout in itself isn't enough. Hikari allows you to set a lot more - check the properties here, we should also configure validationTimeout, loginTimeout and so on
https://github.com/brettwooldridge/HikariCP

Reply:
I have add some more that can be useful but still i haven't understand the need of setting values of: IdleTimeout, keepAliveTimeout, maxLifetime different than defaults

Reply:
ok